prosemirror-jdita/src/schema.ts

```diff
@@ -167,34 +266,73 @@ function defaultTravel(node: typeof BaseNode, parent: typeof BaseNode, next: (no
   return result;
 }
 
+/**
+ * `defaultNodeName` transforms the node `nodeName`
+ * by replacing dashes with underscores
+ *
+ * @remarks
+ * defaultNodeName will return all nodeNames retrieved by `schema()`
+ * and document()
+ *
+ * @example
+ * `media-track` will be transformed to `media_track`
+ *
+ * @param nodeName - The name of the node
+ * @returns A string with the transformed node name
+ */
 export function defaultNodeName(nodeName: string): string {
   return NODE_NAMES[nodeName] || nodeName.replace(/-/g, '_');
 }
 
+/**
+ * `schema` creates a schema for the prosemirror editor
+ * based on the jdita nodes
+ *
+ * @see {@link https://prosemirror.net/docs/ref/#model.SchemaSpec}.
+ *
+ * @returns The Schema Object, describing a schema, as passed to the Schema constructor
+ */
 export function schema(): Schema {
   const done: string[] = [];
+
+  // the schema spec are the nodes and marks
   const spec: SchemaSpec = {
+    // the node types in this schema
     nodes: {
       text: {
         group: 'common_inline all_inline',
         inline: true,
       },
     },
+    // the mark types that exist in this schema
     marks: {},
   }
+
+  // populate the schema spec using the jdita nodes
   function browse(node: string | typeof BaseNode, parent: typeof BaseNode): void {
+    // get the node name
     const nodeName = typeof node === 'string' ? node : node.nodeName;
+
+    // if we have already processed this node then there's no need to process it again
     if (done.indexOf(nodeName) > -1) {
       return;
     }
+    // add the node to the list of done nodes
     done.push(nodeName);
+
+    // do not process the alt or text nodes
+    // TODO: Shouldn't this be done using the node name?
     if (['alt', 'text'].indexOf(node as string) > -1) {
       return;
     }
+
     try {
+      // get the class of the node
       const NodeClass = typeof node === 'string' ? getNodeClassType(node) : node;
```

Comment:
Variable name should not start with a capital letter - please fix!

Reply:
Fixed with 79e3bc8.